module/zfs/arc.c

```diff
@@ -1331,13 +1391,30 @@ arc_buf_lsize(arc_buf_t *buf)
 	return (HDR_GET_LSIZE(buf->b_hdr));
 }
 
+boolean_t
+arc_is_encrypted(arc_buf_t *buf)
```

Comment:
will fix. should i add the one for compression as well while I'm here? Or should that be left to a different PR.